contracts/RMRK/library/RMRKErrors.sol

```diff
@@ -2,59 +2,124 @@
 
 pragma solidity ^0.8.15;
 
+/// @title RMRKErrors
+/// @author RMRK team
+/// @notice A collection of errors used in the RMRK suite
+/// @dev Errors are kept in a centralised file in order to provide a central point of reference and to avoid error
+///  naming collisions due to inheritance
+
+/// Attempting to grant the token to 0x0 address
 error ERC721AddressZeroIsNotaValidOwner();
+/// Attempting to grant approval to the current owner of the token
 error ERC721ApprovalToCurrentOwner();
+/// Attempting to grant approval when not being owner or approved for all should not be permitted
 error ERC721ApproveCallerIsNotOwnerNorApprovedForAll();
+/// Attempting to get approvals for a token owned by 0x0 (considered non-existent)
 error ERC721ApprovedQueryForNonexistentToken();
+/// Attempting to grant approval to self
 error ERC721ApproveToCaller();
```

Comment:
Let's merge this into ERC721ApprovalToCurrentOwner.

Reply:
Could this not be emitted if the caller was approved for all and tried to call approval on specific token for self?

Reply:
What you describe is the case on MultiResource so I guess we need to keep it.
`if (owner == operator) revert ERC721ApproveToCaller();` 
On Nesting and all inheriting implementations, we have an extra level on approvals which is always set to the sender, so there we could use the one I suggested. But it might look inconsistent :thinking: 
`if (_msgSender() == operator) revert ERC721ApproveToCaller();`

Both checks are on `setApprovalForAll`

Reply:
I'd leave these two separate for now and we can address them if we find a good and consistent solution. At least we have a centralised errors file where we can do it :D